File: Problems 2900-2999/2918. Minimum Equal Sum of Two Arrays After Replacing Zeros/solution.py
class Solution:
    def minSum(self, nums1: List[int], nums2: List[int]) -> int:
        # since we're replacing 0 with strictly positive integers
        # the new total must increase (unless no zeroes)
        total1 = 0
        zero1 = 0
        for num in nums1:
            total1 += num
            if num == 0:
                zero1 += 1
        total2 = 0
        zero2 = 0
        for num in nums2:
            total2 += num
            if num == 0:
                zero2 += 1

        # swap, so we just deal with 1 case when totals are different
        if total1 < total2:
            total1, total2 = total2, total1
            zero1, zero2 = zero2, zero1

        # 1. If array sums are equal and if num zeroes are same, then
        # we just add numzeroes to total. if one does not have zeroes
        # but the other does, then return -1. if both have different
        # number of zeroes but both have zeroes, then add
        if total1 == total2:
            if zero1 == zero2:
                return total1 + zero1
            elif zero1 == 0 or zero2 == 0:
                return -1
            else:
                return total1 + max(zero1, zero2)
        else:
            # total1 > total2
            # 2. If one array has greater sum, then ideally we place 1
            # in all zero slots in greater array, then fill zeroes in
            # lesser array with (total1+zero1-total2). edge case
            # where no zeroes in lesser array
            if zero2 == 0 or (total2 + zero2 > total1 and zero1 == 0):
                return -1
            else:
                return max(total1 + zero1, total2 + zero2)

File: Problems 2900-2999/2918. Minimum Equal Sum of Two Arrays After Replacing Zeros/solution.cpp
class Solution {
    public:
        long long minSum(vector<int>& nums1, vector<int>& nums2) {
            long long sum1 = 0;
            long long zero1 = 0;
            for (int num : nums1) {
                sum1 += num;
                if (num == 0) zero1++;
            }
            long long sum2 = 0;
            long long zero2 = 0;
            for (int num : nums2) {
                sum2 += num;
                if (num == 0) zero2++;
            }
    
            // minimum total if replace 0's with 1
            long long minSum1 = sum1 + zero1;
            long long minSum2 = sum2 + zero2;
    
            if ((zero1 == 0 && minSum2 > minSum1) || (zero2 == 0 && minSum1 > minSum2)) {
                return -1;
            }
    
            return max(minSum1, minSum2);
        }
    };